### PR TITLE
Add descheduler 1.24 tests

### PR DESCRIPTION
**Pull request by pravarag**
As part of this change, we are updating the descheduler repo for 1.24 presubmits testing. Previous update: #25093

Related issue: https://github.com/kubernetes-sigs/descheduler/issues/735

### CODE REVIEW

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (damemi) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml:12]: this should be `release-1.24`

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (damemi) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml:16]: 1.24 will be on golang 1.18

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (damemi) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml:51]: same (1.18)

**Review by damemi — COMMENTED** [review]
thanks for starting this @pravarag, noticed some version bumps that we'll need. In the master tests 1.24 should also be bumped to go1.18 too

Holding for now until we're ready for this to merge as part of the release (just so it doesn't merge on accident)
/hold

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (pravarag) [reply config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml]: Thanks for pointing this out, I've updated it with latest push.

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (pravarag) [reply config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml]: Updated.

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml
Comment (pravarag) [reply config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.24.yaml]: Updated.

**Review by damemi — CHANGES_REQUESTED** [review]
@pravarag thanks for making the updates. One last change could you update the `descheder-presubmits-master` tests for 1.24 to use golang1.18? [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L16) and [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L16), and [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L51)

**Comment by pravarag** [comment]
> @pravarag thanks for making the updates. One last change could you update the `descheder-presubmits-master` tests for 1.24 to use golang1.18? [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L16) and [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L16), and [here](https://github.com/kubernetes/test-infra/pull/25833/files#diff-3fa19ca3337a7f655c1acf51b20f718e1762830d6dcd8f93c78ea823637834c0L51)

Updated as suggested.

**Comment by damemi** [comment]
Thanks @pravarag!
/lgtm
/approve

**Comment by JaneLiuL** [comment]
looks need cancel the hold label

**Comment by damemi** [comment]
/hold cancel
thanks @JaneLiuL :)

**Comment by damemi** [comment]
This looks like it broke CI ([sample](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_descheduler/791/pull-descheduler-test-e2e-k8s-master-1-24/1523636388212248576)):
```
+ PATH=/home/prow/go/bin:/go/bin:/usr/local/go/bin:/google-cloud-sdk/bin:/workspace:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/prow/go/src/github.com/kubernetes-sigs/descheduler
+ kind create cluster --image kindest/node:v1.24.0 --config=./hack/kind_config.yaml
Creating cluster "kind" ...
 • Ensuring node image (kindest/node:v1.24.0) 🖼  ...
 ✗ Ensuring node image (kindest/node:v1.24.0) 🖼
ERROR: failed to create cluster: failed to pull image "kindest/node:v1.24.0": command "docker pull kindest/node:v1.24.0" failed with error: exit status 1
```
Looks like the 1.24 node image hasn't been published yet (https://hub.docker.com/r/kindest/node/tags).

**Pull request merged**
